Sources/SKSupport/LineTable.swift

```diff
@@ -123,113 +125,242 @@ extension LineTable {
   }
 }
 
-extension LineTable {
+// MARK: - Position translation
 
-  // MARK: - Position translation
+extension LineTable {
+  // MARK: line:column <-> String.Index
 
-  /// Returns `String.Index` of given logical position.
+  /// Converts the given UTF-16-based `line:column`` position to a `String.Index`.
+  ///
+  /// If the position does not refer to a valid position with in the source file, returns `nil` and logs a fault
+  /// containing the file and line of the caller (from `callerFile` and `callerLine`).
   ///
   /// - parameter line: Line number (zero-based).
   /// - parameter utf16Column: UTF-16 column offset (zero-based).
   @inlinable
-  public func stringIndexOf(line: Int, utf16Column: Int) -> String.Index? {
+  public func stringIndexOf(
+    line: Int,
+    utf16Column: Int,
+    callerFile: StaticString = #fileID,
+    callerLine: UInt = #line
+  ) -> String.Index? {
     guard line < count else {
-      // Line out of range.
+      logger.fault(
+        """
+        Unable to get string index for \(line):\(utf16Column) because line is out of range \
```

Comment:
Probably obvious enough from the caller, but could help to add whether it's utf8 vs utf16 in all these messages. It's not super necessary for the line case, so maybe just add `utf<n>` in front of `column` for the column cases?

Reply:
Good idea 👍🏽